sycl/test-e2e/syclcompat/defs.cpp

```diff
@@ -61,6 +61,19 @@ void test_check_error() {
          SYCLCOMPAT_CHECK_ERROR(runtime_error_throw()));
 }
 
+void test_version() {
+  // Check the composition of the version int
+  assert(SYCLCOMPAT_MAKE_VERSION(1, 1, 1) == 1001001);
+  assert(SYCLCOMPAT_MAKE_VERSION(9, 0, 0) == 9000000);
+
+  // Check the current version is as expected
+  assert(SYCLCOMPAT_MAKE_VERSION(0, 1, 0) == SYCLCOMPAT_VERSION);
```

Comment:
I have mixed feelings about this test, as all our PRs will have to modify both the version and this file.
It's good that it forces us to be explicit about changes in the versioning, but at the same time it feels tedious to test this here.